types/index.d.ts

```diff
@@ -0,0 +1,195 @@
+import { VueConstructor } from 'vue/types/umd';
+
+declare module '@girder/components' {
+  import Vue from 'vue/types';
+  import { Config } from 'vuetify/types';
+  import { AxiosInstance, AxiosResponse } from 'axios';
+
+  type GirderModelType = 'item' | 'folder' | 'file' | 'user';
+
+  export interface GirderModel {
+    name: string;
+    _id: string;
+    _modelType: GirderModelType;
+    parentCollection?: string;
+    parentId?: string;
+    meta: unknown;
+  }
+
+  interface GirderRestClientParams {
+    apiRoot: string;
+    token?: string;
+    axios?: AxiosInstance;
+    authenticateWithCredentials?: boolean;
+    userGirderAuthorizationHeader?: boolean;
+    setLocalCokie?: boolean;
+  }
+
+  class RestClient extends Vue {
+    constructor(params: GirderRestClientParams);
+
+    fetchUser(): Promise<GirderModel>;
+
+    login(username: string, password: string, opt: string | null): Promise<AxiosResponse>;
+
+    logout(): void;
+
+    register(
+      login: string,
+      email: string,
+      firstName: string,
+      lastName: string,
+      password: string,
+      admin: boolean,
+    ): Promise<AxiosResponse>;
+
+    apiRoot: string;
+
+    // inherited from Axios
+    defaults: AxiosInstance['defaults'];
+    interceptors: AxiosInstance['interceptors'];
+    getUri: AxiosInstance['getUri'];
+    request: AxiosInstance['request'];
+    delete: AxiosInstance['delete'];
+    put: AxiosInstance['put'];
+    get: AxiosInstance['get'];
+    head: AxiosInstance['head'];
+    options: AxiosInstance['options'];
+    post: AxiosInstance['post'];
+    patch: AxiosInstance['patch'];
```

Comment:
Is there not some kind of nice way to inherit everything else from AxiosInstance?

Reply:
Not that I'm aware of.  TS doesn't have multiple inheritance, and mixins don't seem to apply to type definitions.